docs/standard/native-interop/tutorial-comwrappers.md

```diff
@@ -384,6 +384,58 @@ Since your `ComWrapper` subclass was designed to support `CreateObjectFlags.Uniq
 (rcw as IDisposable)?.Dispose();
 ```
 
+## COM Activation with `ComWrappers`
+
+The creation of COM objects is typically performed via COM Activation &ndash; a complex scenario outside the scope of this document. In order to provide a conceptual pattern to follow we introduce the [`CoCreateInstance()`][api_cocreateinstance] API, used for COM Activation, and illustrate how it can be used with `ComWrappers`.
+
+Assume you have the following C# code in your application. The below example uses `CoCreateInstance()` to activate a COM class and the built-in COM interop system to marshal the COM instance to the appropriate interface. Note the use of `typeof(I).GUID` is limited to an assert and is a case of using Reflection which can impact if the code is NativeAOT friendly.
+
+```csharp
+public static I ActivateClass<I>(Guid clsid, Guid iid)
+{
+    Debug.Assert(iid == typeof(I).GUID);
+    int hr = CoCreateInstance(ref clsid, IntPtr.Zero, /*CLSCTX_INPROC_SERVER*/ 1, ref iid, out object obj);
+    if (hr < 0)
+    {
+        Marshal.ThrowExceptionForHR(hr);
+    }
+    return (I)obj;
+}
+
+[DllImport("Ole32")]
+extern static int CoCreateInstance(
+    ref Guid rclsid,
+    IntPtr pUnkOuter,
+    int dwClsContext,
+    ref Guid riid,
+    [MarshalAs(UnmanagedType.Interface)] out object ppObj);
+```
+
+Converting the above to use `ComWrappers` involves removing the `MarshalAs(UnmanagedType.Interface)` from the `CoCreateInstance()` P/Invoke and performing the marshalling manually. The [`ComWrappers.GetOrRegisterObjectForComInstance()`][api_comwrappers_getorregisterobjectforcominstance] could also be used to abstract away factory style functions and activate from within a class constructor.
+
+```csharp
+static ComWrappers s_ComWrappers = ...;
+
+public static I ActivateClass<I>(Guid clsid, Guid iid)
+{
+    Debug.Assert(iid == typeof(I).GUID);
+    int hr = CoCreateInstance(ref clsid, IntPtr.Zero, /*CLSCTX_INPROC_SERVER*/ 1, ref iid, out IntPtr obj);
+    if (hr < 0)
+    {
+        Marshal.ThrowExceptionForHR(hr);
+    }
+    return (I)s_ComWrappers.GetOrCreateObjectForComInstance(obj, CreateObjectFlags.None);
+}
+
+[DllImport("Ole32")]
+extern static int CoCreateInstance(
+    ref Guid rclsid,
+    IntPtr pUnkOuter,
+    int dwClsContext,
+    ref Guid riid,
+    out IntPtr ppObj);
+```
+
```

Comment:
```suggestion

It is also possible to abstract away factory-style functions like `ActivateClass<I>` by including the activation logic in the class constructor for a Native Object Wrapper. The constructor can use the [`ComWrappers.GetOrRegisterObjectForComInstance()`][api_comwrappers_getorregisterobjectforcominstance] API to associate the newly constructed managed object with the activated COM instance.
```

Reply:
It took me a bit to see what the original sentence was getting at - tried to rephrase a bit. Feel free to take it or not.